### DISCUSSION TITLE
Sample files

### DISCUSSION THREAD

**Discussion by pragma37** [bnpr/Malt#94] [Showcase]
Here we can share example blend files.

Feel free to submit your own!

**Comment by pragma37** [comment]
*(This sample is outdated)*

A basic example showing the 4 types of Malt node trees. (Previously available on my Patreon)

[nodes-example.zip](https://github.com/bnpr/Malt/files/6726809/nodes-example.zip)

![imagen](https://user-images.githubusercontent.com/36610999/123649215-e0b2b900-d829-11eb-9fe7-4a4fcd1c95ff.png)

**Comment by pragma37** [comment]
*(This sample is outdated)*

How to setup Custom AOVs:

[Custom AOV example.zip](https://github.com/bnpr/Malt/files/7001337/Custom.AOV.example.zip)

![imagen](https://user-images.githubusercontent.com/36610999/129757640-d5f31747-0fc2-4ca5-a279-b662da13de5e.png)
```python
from Malt.Pipelines.NPR_Pipeline.NPR_Pipeline_Nodes import *

class Custom_AOV_Pipeline (NPR_Pipeline_Nodes):

    def get_mesh_shader_custom_outputs(self):
        # MAX CUSTOM OUTPUTS is 5
        # Use GL_RGBA32UI for packed textures 
        # Use GL_RGBA16F or GL_RGBA32F for standard textures 
        return {
            'AO' : GL_RGBA16F,
            'Unlit' : GL_RGBA16F,
            'True Normal' : GL_RGBA16F,
            'Packed Masks ABCD' : GL_RGBA32UI,
        }
    
    def get_render_layer_custom_outputs(self):
        # NO MAX OUTPUTS
        # Always use GL_RGBA16F or GL_RGBA32F         
        return {
            'AO' : GL_RGBA16F,
            'Unlit' : GL_RGBA16F,
            'True Normal' : GL_RGBA16F,
            'Mask A' : GL_RGBA16F,
            'Mask B' : GL_RGBA16F,
            'Mask C' : GL_RGBA16F,
            'Mask D' : GL_RGBA16F,
        }

PIPELINE = Custom_AOV_Pipeline
```

**Comment by pragma37** [comment]
Malt 1.0 Material Examples

[material_examples_2.zip](https://github.com/bnpr/Malt/files/9465452/material_examples_2.zip) | [material_examples.zip (Old)](https://github.com/bnpr/Malt/files/8008818/material_examples.zip)
![material-examples](https://user-images.githubusercontent.com/36610999/152656985-98ffe8f5-7174-4b1e-984f-14fed65098b4.png)

**Reply by ghost** [reply]
How to use this files?

**Reply by pragma37** [reply]
For this one, you need the [1.0 preview release](https://github.com/bnpr/Malt/discussions/231) installed.
And then just open the file.

**Reply by datOne23** [reply]
I'm still figuring out how to use toon and diffuse shading with a custom base and shadow color

**Reply by pragma37** [reply]
Hey @datOne23, if you open a new question detailing what do you want to achieve, I can point you in the right direction.
https://github.com/bnpr/Malt/discussions/new?category=q-a

**Reply by panthavma** [reply]
Hello, current version of this file doesn't work with the latest MALT release.
- The label field in the malt tree (containing 'mesh', 'light', etc). Setting it to 'Mesh Shader' works.
- I can't set the render pipeline in the malt setting to a node tree
- In the meantime, all the spheres are yellow even with a set material, or one from the shader examples.

(I don't know if this is the right place to report this)

**Reply by pragma37** [reply]
@panthavma This is not a bug. You can find the solution two comments above. 
You need the [1.0 preview version](https://github.com/bnpr/Malt/discussions/231) to run this sample file.

**Reply by panthavma** [reply]
Ah my bad, I thought the latest release was 1.0
Thanks for the answer ! :)
